src/site/resources/js/checkstyle.js

```diff
@@ -25,3 +25,21 @@ window.addEventListener("scroll", function () {
         scrollButton.style.display = "none";
     }
 });
+
+function setBodyColumnMargin() {
+    const leftColumn = document.querySelector('#leftColumn');
+    const bodyColumn = document.querySelector('#bodyColumn');
+
+    // If in mobile view use margin as defined in site.css
+    if (window.innerWidth < 823) {
+        bodyColumn.style.marginLeft = '1.5em';
+        return;
+    }
+
+    // Else calculate margin based on left column width
+    const leftColumnWidth = leftColumn.offsetWidth;
+    bodyColumn.style.marginLeft = `${leftColumnWidth + 27}px`;
+}
+
+window.addEventListener('load', setBodyColumnMargin);
+window.addEventListener('resize', setBodyColumnMargin);
```

Comment:
> And why this? What was wrong with the old CSS handling?

Old CSS handling relied on a fixed `margin-left` to move the main content from the menu. Now that the width of the menu can change, the left margin needs to be calculated based on that. CSS cannot deal with such dynamic calculations to my knowledge.
https://github.com/checkstyle/checkstyle/blob/3d8ed42ab60c0399550b3e7d54ca9dd767583101/src/site/resources/css/site.css#L96-L102

Reply:
@romani @nrmancuso This will break the page when Javascript is disabled.
![image](https://github.com/checkstyle/checkstyle/assets/5427943/9665ec36-f42e-49f2-b1a2-6e013e56b00a)

Reply:
@romani @nrmancuso This will break the page when Javascript is disabled.
![image](https://github.com/checkstyle/checkstyle/assets/5427943/c9714d98-d288-4da0-8d6a-16da8a0d9c02)

Reply:
20 years ago disabled JavaScript was normal, but today browser without JavaScript is not working application :).

If fix is easy, we can consider to support non JavaScript browser.

Reply:
Javascript disable is still normal today as a security function. https://noscript.net/ is still being actively developed and used, especially in the Tor browser.

Reply:
If we end up in problem to make it for secure mode of browser, I recommend to merge as is, it will work for 90% user, and create issue to experiment of only css solution.

I am not expert here, I can not lead on how to do this.

Reply:
> @stoyanK7 Something like https://stackoverflow.com/a/38150661/1016482 won't work?

I tried but couldn't make it work since we are trying to set a `margin`, not a `width`. I found https://stackoverflow.com/questions/49286748/css-set-the-margin-of-a-div-equal-to-another-divs-width but it uses predefined `width` which is not applicable if we want to stretch the menu based on the content inside.

The only viable non-javascript solution I can think of is using grid. Though we are dropping some browser compatibility for that - browser versions before 2015-2012: https://caniuse.com/css-grid

Reply:
@rnveach , I think we need merge as is, and create new issue to improve, but we will need your experience in html to fix it.